### PR TITLE
React: modified gene search to display multiple ensembl IDs per gene name

### LINKED ISSUE #175: React: modify dropdown for gene name with multiple ensembl IDs
Currently, our dropdown only supports 1-to-1 relationship for 1 gene name that corresponds to 1 ensembl ID. There genes, such as ABHD16A, that has multiple ensembl IDs. We should give users warning about genes with multiple IDs and prompts them to choose the appropriate ID.

### PR DESCRIPTION
**Pull request by hannahle**

### CODE REVIEW

File: react/src/components/GeneSearch.tsx [hunk 1]
```diff
@@ -40,6 +42,9 @@ interface GeneSearchProps {
     onChange: (geneName: string) => void;
 }
 
+/* Typeguard */
+const isMultipleGenes = (arg: Gene | Gene[]): arg is Gene[] => (arg as Gene[]).length >= 0;
```
Comment (kevinlul) [inline react/src/components/GeneSearch.tsx]: `Array.isArray` is probably a better check

File: react/src/components/GeneSearch.tsx [hunk 2]
```diff
@@ -57,16 +62,35 @@ const GeneSearch: React.FC<GeneSearchProps> = ({ assembly, geneName, onChange, o
...
+                            name: hit.symbol.toUpperCase(),
+                            ensemblId: e.gene,
+                            position: getPosition(assembly, {
+                                genomic_pos: hit.genomic_pos[eid],
+                                genomic_pos_hg19: hit.genomic_pos_hg19[eid],
+                            }),
+                        },
+                        id: i + eid,
+                        label: isMultipleGenes(hit.ensembl) ? e.gene : hit.symbol.toUpperCase(),
+                    }))
```
Comment (kevinlul) [inline react/src/components/GeneSearch.tsx]: You could do this in the prior map call to only loop through the hits once, unless you plan to reuse the separate functions.

File: react/src/components/GeneSearch.tsx (hunk 1 above)
Comment (hannahle) [reply react/src/components/GeneSearch.tsx]: Good point!

File: react/src/components/GeneSearch.tsx (hunk 2 above)
Comment (hannahle) [reply react/src/components/GeneSearch.tsx]: Sounds good!

**Pull request merged**